### PR TITLE
modify examples to detect websocket uri based on request

### PR DESCRIPTION
**Pull request by stepheny**
select correspongding scheme ws or wss based on the request scheme
allow different hostname and port bombination
especially helpful uwsgi listening on socket with nginx reverse proxy

### CODE REVIEW

File: examples/chat-gevent/chat.py [hunk 1]
```diff
@@ -11,7 +11,11 @@
 
 @app.route('/')
 def index():
-    return render_template('index.html')
+    uri = '%s://%s/%s' % \
+        (request.environ['wsgi.url_scheme'] == 'https' and 'wss' or 'ws', \
```
Comment (Ivoz) [inline examples/chat-gevent/chat.py]: This should use a more readable syntax than inline `and` and `or`, e.g:

    scheme = 'wss' if (request.environ['wsgi.url_scheme'] == 'https') else 'ws'
    uri = '%s://%s/websocket' % (scheme, request.headers['host'])

**Comment by Ivoz** [comment]
Could this not also just be done in the javascript, client side?

    (location.protocol === "https:") ? "wss" : "ws";

**Comment by zeekay** [comment]
That's what I do, generally.

—
Zach Kelling
$ curl monoid.io/say-hi | python
zk@monoid.io / (913) 777-9708

**Comment by stepheny** [comment]
I prefer the javascript side too.

**Pull request closed without merging**